starknet-foundry/crates/forge/src/lib.rs

```diff
@@ -70,25 +70,35 @@ struct TestsFromFile {
 }
 
 fn collect_tests_from_directory(
-    input_path: &Utf8PathBuf,
+    package_path: &Utf8PathBuf,
+    lib_path: &Utf8PathBuf,
     linked_libraries: &Option<Vec<LinkedLibrary>>,
     corelib_path: Option<&Utf8PathBuf>,
     runner_config: &RunnerConfig,
 ) -> Result<Vec<TestsFromFile>> {
-    let test_files = find_cairo_files_in_directory(input_path)?;
+    let test_files = find_cairo_root_files_in_directory(package_path, lib_path)?;
     internal_collect_tests(
-        input_path,
+        package_path,
         linked_libraries,
         &test_files,
         corelib_path,
         runner_config,
     )
 }
 
-fn find_cairo_files_in_directory(input_path: &Utf8PathBuf) -> Result<Vec<Utf8PathBuf>> {
-    let mut test_files: Vec<Utf8PathBuf> = vec![];
-
-    for entry in WalkDir::new(input_path).sort_by(|a, b| a.file_name().cmp(b.file_name())) {
+fn find_cairo_root_files_in_directory(
+    input_path: &Utf8PathBuf,
+    lib_path: &Utf8PathBuf,
+) -> Result<Vec<Utf8PathBuf>> {
+    let mut test_files: Vec<Utf8PathBuf> = vec![lib_path.clone()];
+    let default_src_path = &input_path.join(Utf8PathBuf::from("/src"));
+    let src_path = lib_path.parent().unwrap_or(default_src_path);
```

Comment:
I would just bubble this error if there is no parent instead of using a default value.